cf-monitord/mon_network.c

```diff
@@ -429,7 +430,9 @@ void MonNetworkGatherData(double *cf_this)
         }
 
         SetNetworkEntropyClasses(CanonifyName(ECGSOCKS[i].name), "in", in[i]);
+        const mode_t old_umask = SetUmask(0077);
         RawSaveItemList(in[i], vbuff, NewLineMode_Unix);
+        RestoreUmask(old_umask);
```

Comment:
I wonder if it wouldn't be better to put the SetUmask(0077) in RawSaveItemList. I think there are a few more places this function is called (you already got MonProcessesGatherData).

```
core/cf-monitord/mon_processes.c MonProcessesGatherData 64 RawSaveItemList(userList, vbuff, NewLineMode_Unix);
core/libpromises/processes_select.c LoadProcessTable 1665 RawSaveItemList(PROCESSTABLE, vbuff, NewLineMode_Unix);
core/libpromises/processes_select.c LoadProcessTable 1710 RawSaveItemList(rootprocs, vbuff, NewLineMode_Unix);
core/libpromises/processes_select.c LoadProcessTable 1714 RawSaveItemList(otherprocs, vbuff, NewLineMode_Unix);
enterprise/libcfenterprise/win_ps.c LoadProcessTable 70 RawSaveItemList(*procdata, buf, NewLineMode_Unix);
enterprise/libcfenterprise/win_ps.c LoadProcessTable 88 RawSaveItemList(systemprocs, buf, NewLineMode_Unix);
enterprise/libcfenterprise/win_ps.c LoadProcessTable 92 RawSaveItemList(otherprocs, buf, NewLineMode_Unix);
```

Reply:
Granted that would be a bit more complicated since there are multiple points of return in RawSaveItemList() so maybe just add SetUmask()/RestoreUmask() around the other bits above that I mentioned. Or if Karl's changes are soon to be merged, let it go?

Reply:
I don't think it's a good idea to enforce permissions in `RawSaveItemList()` which is a generic function saving an item list.

Reply:
We will enforce permissions by default in safe_fopen soon which is used by RawSaveItemList() right? 

https://github.com/cfengine/core/pull/3728/files#diff-273429e7baa188e67488632c298ac190R488 safe_fopen()

https://github.com/cfengine/core/pull/3728/files#diff-cec9a2415beb622eecf3d69f043d5181R52 (CF_PERMS_DEFAULT)

So probably it doesn't matter much. Leave as-is, and now it's merged anyway. So :trophy:  :p

Reply:
I agree @vpodzime that it shouldn't be in a generic function. Was just looking for a way to reduce the amount of changes needed now and after Karl's PR is merged. :+1: